OpenRA.Mods.Common/Traits/AmmoPool.cs

```diff
@@ -106,26 +106,40 @@ void INotifyAttack.Attacking(Actor self, Target target, Armament a, Barrel barre
 
 		void INotifyAttack.PreparingAttack(Actor self, Target target, Armament a, Barrel barrel) { }
 
-		void ITick.Tick(Actor self)
+		public virtual void Reload(Actor self, int reloadDelay, int reloadCount, string sound, bool needsToBeDocked, bool resetOnFire = false)
```

Comment:
This isn't great for separation of concerns – the `AmmoPool` really shouldn't know anything about `needsToBeDocked` `RemainingTicks` or `resetOnFire` (and also maybe not `sound`).  Can you please pull all that logic up into `Rearm` and `ReloadAmmoPool`?  `ReloadAmmoPool` can implement `INotifyAttack` to find out when to reset its counter, and then both can call `Reload(Actor self, int count)` when it is time to unconditionally increment the ammo pool count.